library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -256,21 +262,20 @@ def all
           Yast::Path.new(".target.bash_output"),
           format(LIST_SNAPSHOTS_CMD, root: target_root.shellescape)
         )
-        lines = out["stdout"].lines.grep(VALID_LINE_REGEX) # relevant lines from output.
+
         log.info("Retrieving snapshots list: #{LIST_SNAPSHOTS_CMD} returned: #{out}")
-        lines.each_with_object([]) do |line, snapshots|
-          data = line.split("|").map(&:strip)
-          next if data[0] == "0" # Ignores 'current' snapshot (id = 0) because it's not a real snapshot
-
-          begin
-            timestamp = DateTime.parse(data[3])
-          rescue ArgumentError
-            log.warn("Error when parsing date/time: #{timestamp}")
-            timestamp = nil
-          end
-          previous_number = (data[2] == "") ? nil : data[2].to_i
-          snapshots << new(data[0].to_i, data[1].to_sym, previous_number, timestamp,
-            data[4], data[5].to_sym, data[6])
+
+        csv = CSV.parse(out["stdout"], headers: true, converters: [:date_time, :numeric])
```

Comment:
what about errors? E.g. old code handle data_time conversion error, what happen here if something is malformed?

Reply:
Oh yes. That check is missing now. Thanks!